star-sys/STAR/source/stitchAlignToTranscript.cpp

```diff
@@ -5,6 +5,8 @@
 #include "binarySearch2.h"
 // #include "stitchGapIndel.cpp"
 
+Transcript trExtend;
```

Comment:
@adam-azarchs profiling showed a lot of time was spent initializing this object, am hoping by declaring it once at the top level and then callling `.reset()` we'll avoid lots of memory allocations, does this seem kosher to you?

Reply:
No, it doesn't.  This can absolutely be run in a multithreaded context where this would cause errors.

Also, `reset`  definitely doesn't actually reset the transcript object to its initial state.